### PR TITLE
Support config file for server configuration

### LINKED ISSUE #201: [Feature] Use the config file for configuration
**Feature (What you would like to be added):**
- [ ] Introduce new flag `--config-file` which accepts the config file path.
- [ ] Use this json/yaml config file to configure etcd-backup-restore server.
- [ ] Also, centralise the validation for all flags

**Motivation (Why is this needed?):**
Currently there are too many flags for `etcdbrctl server` command. Configuring each flag on for command lengthens the command line drastically. Instead its better to have it in single config file with section for different correlated flags.

**Approach/Hint to the implement solution (optional):**
- Use library like https://github.com/ghodss/yaml or https://github.com/spf13/viper to parse the config file.

### PR DESCRIPTION
**Pull request by swapnilgm**
Signed-off-by: Swapnil Mhamane <swapnil.mhamane@sap.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #201 

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Configuring backup-restore server using config file is now supported.
```

### CODE REVIEW

File: example/00-backup-restore-server-config.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  # insecureSkipVerify: true
+  # certFile: "ssl/etcd/tls.crt"
+  # keyFile: "ssl/etcd/tls.key"
+  # caFile: "ssl/etcd/ca.crt"
+
+serverConfig:
+  port: 8080
+  # enableProfiling: true
+  # server-cert: "ssl/etcdbr/tls.crt"
+  # server-key: "ssl/etcdbr/tls.crt"
```
Comment (shreyas-s-rao) [inline example/00-backup-restore-server-config.yaml]:
```suggestion
  # server-key: "ssl/etcdbr/tls.key"
```

File: pkg/snapshot/snapshotter/init.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	cron "github.com/robfig/cron/v3"
+	"github.com/sirupsen/logrus"
+	flag "github.com/spf13/pflag"
+)
+
+// NewSnapshotterConfig returns the snapshotter config.
+func NewSnapshotterConfig() *Config {
+	return &Config{
+		FullSnapshotSchedule:     "0 */1 * * *",
+		DeltaSnapshotPeriod:      wrappers.Duration{Duration: 20 * time.Second},
```
[the remark this answers is not in the document]
Comment (swapnilgm) [reply pkg/snapshot/snapshotter/init.go]: Done

File: pkg/snapshot/snapshotter/init.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		DeltaSnapshotMemoryLimit: 10 * 1024 * 1024, //10Mib
```
[the remark this answers is not in the document]
Comment (swapnilgm) [reply pkg/snapshot/snapshotter/init.go]: Done

File: pkg/snapshot/snapshotter/init.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		GarbageCollectionPeriod:  wrappers.Duration{Duration: time.Minute},
+		GarbageCollectionPolicy:  GarbageCollectionPolicyExponential,
+		MaxBackups:               7,
```
[the remark this answers is not in the document]
Comment (swapnilgm) [reply pkg/snapshot/snapshotter/init.go]: Done

File: pkg/snapshot/restorer/init.go [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/coreos/etcd/pkg/types"
+	flag "github.com/spf13/pflag"
+)
+
+const (
```
Comment (shreyas-s-rao) [inline pkg/snapshot/restorer/init.go]: For consistency, bring these constants to `pkg/snapshot/restorer/types.go`. Other than this, LGTM 👍🏼

**Pull request merged**